File: src/main/java/org/mitallast/queue/crdt/commutative/CmRDT.java
package org.mitallast.queue.crdt.commutative;

import org.mitallast.queue.common.codec.Message;
import org.mitallast.queue.crdt.Crdt;

public interface CmRDT extends Crdt {

    interface SourceUpdate extends Message {}

    interface DownstreamUpdate extends Message {}

    void sourceUpdate(SourceUpdate update);

    void downstreamUpdate(DownstreamUpdate update);
}

File: src/main/java/org/mitallast/queue/crdt/commutative/CmRDT.kt
package org.mitallast.queue.crdt.commutative

import org.mitallast.queue.common.codec.Message
import org.mitallast.queue.crdt.Crdt

interface CmRDT : Crdt {

    interface SourceUpdate : Message

    interface DownstreamUpdate : Message

    fun sourceUpdate(update: SourceUpdate)

    fun downstreamUpdate(update: DownstreamUpdate)
}
